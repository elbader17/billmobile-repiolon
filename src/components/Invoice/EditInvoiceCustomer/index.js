import { connect } from 'react-redux';
import NewCustomer from '../../Customer/NewCustomer/Component';
import { updateFiscalIdentity } from '../../../app/fiscal_identity/actions';

const mapStateToProps = state => ({
  fiscalIdentity: state.invoices.currentInvoice.fiscalIdentity,
  invoiceId: state.invoices.currentInvoice.id,
});

const mapDispatchToProps = (dispatch) => {
  return {
    saveFiscalIdentity: (name, identification, category, id, address, city) => {
      const resource = {
        id,
        name,
        identification,
        category, 
        business_address: address, 
        city
      }
      return dispatch(updateFiscalIdentity(resource))
    },
    type: 'invoice'
  };
}

const component = connect(
  mapStateToProps,
  mapDispatchToProps,
)(NewCustomer);

export default component;