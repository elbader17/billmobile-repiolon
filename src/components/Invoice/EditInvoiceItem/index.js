import { connect } from 'react-redux';
import NewItem from '../../Item/NewItem/Component';
import { updateInvoiceItem } from '../../../app/invoice_items/actions';

const mapStateToProps = state => ({});

function mapDispatchToProps(dispatch) {
  return {
    saveItem: (attributes) => {
      const {id, category, name, price} = attributes;
      return dispatch(updateInvoiceItem(id, {category, name, price}))
    },
    type: 'invoice'
  };
}

const component = connect(
  mapStateToProps,
  mapDispatchToProps,
)(NewItem);

export default component;