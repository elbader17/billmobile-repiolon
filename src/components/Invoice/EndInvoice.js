import React from 'react';
import { View, Text, TouchableOpacity, ActivityIndicator, Alert } from 'react-native';
import { Button } from "react-native-elements";
import { IconAddCustomer, IconMore, IconCustomer2, IconX } from '../../constants/icons';
import { COLORS } from '../../constants/colors';
import style from './style';

class EndInvoice extends React.Component {

    constructor(props) {
      super(props);
      this.state = {
        loadingItems: false
      }
    }

    componentWillMount() {
      this.props.getInvoiceItems()
        .then(() => this.setState({loadingItems: true}))
    }

    cancelCbte = () => {
        Alert.alert(
          'Se Cancelará el Comprobante Actual','¿Está Seguro?',
          [
            { //Press Cancel
              onPress: () => console.log('Cancel Cbte'),
              text: 'Cancelar',
              style: 'cancel',
            },
            { //Press OK
              text: 'Borrar', onPress: () => {
                this.props.setRenderEndInvoice(false);
                this.props.setRenderButtonsNewDraft(true);
                this.props.resetCurrentInvoice(),
                this.props.setStateCurrentInvoice()
              }//End onPress
            },
          ],
          {cancelable: false},
        );
    }  

    renderLoading = () => (
      <View style={{marginBottom: 7}}>
        <ActivityIndicator size="large" color={COLORS.blue}/>
      </View>
    );
    
    render() {
      const displayRenderCustomer = this.props.showCustomer ? 'flex' : 'none';
      const displayButtonAddCustomer = (this.props.showCustomer || this.props.loadingFC) ? 'none': 'flex';
        return (
          <View style={style.containerInitEndInvoice}>
            <View style={style.containerBody}>
          
              <Text style={style.textRegular12GrayDark}>
                Datos del Receptor
              </Text>
              <View style={style.containerBoxInvoice}>
                <View style={{display: displayRenderCustomer}}>
                {this.props.loadingFC ? this.renderLoading() : this.props.renderCustomer() }
                </View>
                <View style={{display: displayButtonAddCustomer}}>
                  <View style={style.inLineSpaceAround}>
                    <Button
                      title=' Consumidor Final'
                      testID='addCustomer'
                      TouchableComponent={TouchableOpacity}
                      icon={<IconCustomer2 size={15} color={COLORS.blueLight}/>}
                      onPress={ () => this.props.addFinalConsumer() }
                      buttonStyle={style.buttonAddFinalConsumer}
                      titleStyle={style.textRegular14White}
                    />
                    <Button
                      title=' Otro Cliente'
                      testID='addCustomer'
                      TouchableComponent={TouchableOpacity}
                      icon={<IconAddCustomer color={COLORS.blueLight}/>}
                      onPress={ this.props.navigateClient }
                      buttonStyle={style.buttonAddCustomer}
                      titleStyle={style.textRegular14White}
                    /> 
                  </View>
                </View>
              </View>
    
              <Text style={style.textRegular12GrayDark}>
                Detalle Producto / Servicio
              </Text>
              <View style={style.containerBoxInvoice}>
                <View>
                  {this.state.loadingItems ? this.props.renderViewItemsAdd() : this.renderLoading()}
                </View> 
                <View style={{alignItems: 'center'}}>
                  <Button
                    title=' Agregar Producto/Servicio'
                    TouchableComponent={TouchableOpacity}
                    onPress={ this.props.navigateAddItems }
                    icon={IconMore}
                    buttonStyle={style.buttonAdd}  
                    titleStyle={ style.textRegular14White }
                    disabledStyle={style.buttonAddDisabled}
                    disabledTitleStyle = { style.textRegular16GrayLight }
                  />
                </View>   
              </View>
              <View>
                <Button
                  title='  Cancelar Comprobante  '
                  icon={<IconX color={'red'} size={21} />}
                  TouchableComponent={TouchableOpacity}
                  onPress={this.cancelCbte}
                  buttonStyle={style.buttonCancelInvoice}
                  titleStyle={style.textRegular14Red}
                />
                <Button
                  title='  Volver Atrás  '
                  TouchableComponent={TouchableOpacity}
                  onPress={() => {
                    this.props.setRenderEndInvoice(false);
                    this.props.setRenderInitInvoice(true)
                  }}
                  buttonStyle={style.buttonBackInvoice}
                  titleStyle={style.textRegular14White}
                />
              </View>
            </View>

            <View style={style.containerFooter}>
              <Button
                title='Continuar'
                TouchableComponent={TouchableOpacity}
                onPress={this.props.navigateToSummaryInvoice}
                buttonStyle={style.buttonContinue}
                titleStyle={style.textBold16White}
                loading={this.props.loadingContinue}
              />
            </View>
    
          </View>
        )
      }
}

export default EndInvoice;