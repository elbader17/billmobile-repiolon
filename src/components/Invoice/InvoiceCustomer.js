import React from 'react';
import { View, TextInput, ScrollView } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import { Button } from "react-native-elements";
import PropTypes from 'prop-types';
import InvoiceListCustomer from './InvoiceListCustomers';
import { validateDni } from '../../utils/validations'
import style from './style';

const InvoiceCustomer = props => {
  const iconEnabled = <Icon name="md-return-right" size={18} color="#EE6123"/>;
  const iconDisabled = <Icon name="md-return-right" size={18} color="#858585"/>;
  if (!props.finalConsumer) {
    return (
      <ScrollView>
        <View style={style.listCustomer}>
          <InvoiceListCustomer
            identityCustomer={props.fiscalIdentity}
          />
        </View>
      </ScrollView>
    );
  }else {
    return (
      <View style={style.containerFinalConsumer}>
        <View style={[style.inLineSpaceBetween, {alignItems: 'center'}]}>
          <TextInput
            placeholder="DNI"
            placeholderTextColor={'#cecece'}
            onChangeText={props.setFinalConsumer}
            style={[style.textRegular18GrayDark,style.inputDNICustomer]}
            keyboardType='numeric'
          />
          <Button
            icon={ validateDni(props.identity) ? iconEnabled : iconDisabled  }
            onPress={ props.addFinalConsumer }
            buttonStyle={ style.buttonCheck }
            disabledStyle= { style.buttonCheckDisabled }
            loading = { props.loading }
            disabled = { props.loading || !validateDni(props.identity) }
          />
        </View>
      </View>
    );
  }
}

InvoiceCustomer.propTypes = {
  finalConsumer: PropTypes.bool,
  setFinalConsumer: PropTypes.func,
  addFinalConsumer: PropTypes.func,
  loading: PropTypes.bool,
  identity: PropTypes.string
};

export default InvoiceCustomer;