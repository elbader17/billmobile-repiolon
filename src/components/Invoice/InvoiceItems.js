import React from 'react';
import { View, Text, ScrollView, TouchableOpacity } from 'react-native';
import { Button } from "react-native-elements";
import Icon from 'react-native-vector-icons/Feather';
import PropTypes from 'prop-types';
import { COLORS } from '../../constants/colors';
import style from './style';

const InvoiceItems = props => {
  console.log(props.items);
  return (
    <View style={[style.containerItemsInvoice,style.inColumnSpaceBetween]}>
      <ScrollView style={style.styleScroll}>
        {props.items.map((item, index) => (
          <View key={index}>
            <View style={[style.inLineSpaceBetween, style.line]}>
              <View style={style.boxItems1}>
                <Text style={style.textLight16GrayDark} numberOfLines={1}>
                  {item.name}
                </Text>
              </View>
              <View style={[style.inLine, style.boxItems2]}>
                <Button
                  title={`+${item.quantity}`}
                  TouchableComponent={TouchableOpacity}
                  onPress={props.incrementQuantity}
                  buttonStyle={style.buttonCantProduct}
                  titleStyle={style.textRegular12BlueMedium}
                />
                <Button
                  icon = {<Icon name="x" size={13} color={COLORS.blueMedium} />}
                  TouchableComponent={TouchableOpacity}
                  onPress={() => props.deleteItem(props.invoiceId)}
                  buttonStyle={[style.buttonCantProduct, {marginLeft: 5}]}
                  titleStyle={style.textRegular12BlueMedium}
                />
              </View>
              <View style={style.boxItems3}>
                <Text style={style.textLight16GrayDark}>
                  ${item.price}
                </Text>
              </View>
            </View>
          </View>
        ))}
      </ScrollView>

      <View style={style.boxItemsInvoiceTotal}>
        <View style={style.inLineSpaceBetween}>
          <Text style={style.textLight18BlueMedium}>Total</Text>
          <Text style={style.textLight18BlueMedium}>$ {props.total}</Text>
        </View>
      </View>
    </View>
  );
};

InvoiceItems.propTypes = {
  items: PropTypes.array,
  price: PropTypes.number,
  onPress: PropTypes.func
};

export default InvoiceItems;