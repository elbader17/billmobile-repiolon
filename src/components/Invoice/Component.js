import React from 'react';
import { View, Text, Modal, TouchableOpacity, Alert } from 'react-native';
import DateTimePicker from 'react-native-modal-datetime-picker';
import LinearGradient from 'react-native-linear-gradient';
import { Button } from "react-native-elements";
import { showMessage } from "react-native-flash-message";
import ModalVoucherTYpe from './ModalVoucherType';
import InvoiceCustomer from './InvoiceCustomer';
import InvoiceItems from './InvoiceItems';
import { validateData, validateDni } from '../../utils/validations';
import { presentInvoiceDate } from '../../utils/date';
import { 
  messageItemsIncomplete, 
  messageCustomerIncomplete 
} from '../../utils/messagesNotifications';
import { IconDocument, IconAddCustomer, IconMore } from '../../constants/icons';
import { GRADIANTBLUE2, COLORGBL } from '../../constants/colors';
import { VOUCHER_TYPES } from '../../constants/invoice';
import { XY } from '../../constants/gradientCoord';
import style from './style';

class Invoice extends React.Component {

  constructor(props) {
    super(props);
    const { voucherType, fiscalIdentity } = this.props;
    this.state = {
      voucherType,
      invoiceDate: new Date(this.props.invoiceDate),
      fcIdentification: undefined,
      showCustomer: fiscalIdentity.name != '',
      isDateTimePickerVisible: false,
      isDateTimeVisible:false,
      modalVisible: false, //Show Modal Voucher Type
      loading: false, //For buttons
      quantity: 1, //Cant items
      validIdentity: true,
    }
  }

  static navigationOptions = () => {
    return {
      title: 'Generación de Comprobante',
      headerTransparent: true,
      headerStyle: style.headerStyle,
      headerTitleStyle: style.headerText,
      headerTintColor: 'white',
      headerLeft: IconDocument
    }  
  };

  setModalVisible = visible => this.setState({modalVisible: visible});
  showDateTimePicker = () => this.setState({ isDateTimePickerVisible: true });
  hideDateTimePicker = () => this.setState({ isDateTimePickerVisible: false });
  setFcIdentification = value => this.setState({ fcIdentification: value, validIdentity: true });
  setShowCustomer = value => this.setState({ showCustomer: value });
  setLoading = (bool) => this.setState({ loading: bool });
  
  navigateClient = () => {
    this.setShowCustomer(true);
    this.props.navigation.navigate('ListInvoiceCustomer');
  }
  navigateAddItems = () => this.props.navigation.navigate('ListInvoiceItem');
  navigateToSummaryInvoice = () => {
    const { showCustomer } = this.state;
    const { fiscalIdentity, items } = this.props;
    if (validateData(fiscalIdentity.name, items.length)){
      if (!showCustomer) {
        const customer = fiscalIdentity.name === 'fc' ? fiscalIdentity.identification : fiscalIdentity.name;
        const title = fiscalIdentity.name === 'fc' ? 'Consumidor Final: ' : 'Cliente: ' ;
        Alert.alert(
          title + customer,'El comprobante se confeccionará con éste Cliente',
          [
            { text: 'Cambiar', style: 'cancel' },
            { text: 'Continuar', onPress: () => this.props.navigation.navigate('InvoiceSummary') },
          ], {cancelable: false},
        );
      } 
      else this.props.navigation.navigate('InvoiceSummary');
    } else 
      if (fiscalIdentity.name != '') 
        showMessage(messageItemsIncomplete);
      else showMessage(messageCustomerIncomplete)
  }

  updateInvoiceItemQuantity = (invoiceItemId, quantity) => {
    const { updateInvoiceItemQuantity } = this.props;
    updateInvoiceItemQuantity(invoiceItemId, quantity);
  }
  
  handleDatePicked = (date) => {
    this.setState({ invoiceDate: date });
    this.hideDateTimePicker();
    this.createOrUpdateInvoice({ invoiceDate: date });
  }

  createOrUpdateInvoice = (values) => {
    const { updateInvoice, createInvoice, invoiceId } = this.props;
    const { invoiceDate, voucherType } = this.state ;
    if (invoiceId != null) updateInvoice(values);
    else createInvoice(invoiceDate, voucherType);
  };

  presentVoucherType = () => {
    const { voucherType } = this.state;
    return VOUCHER_TYPES.find((v) => v.value === voucherType).label;
  }

  selectionVoucher = (voucherType) => {
    this.setModalVisible(!this.state.modalVisible);
    this.setState({voucherType: voucherType.value });
  }

  incrementQuantity = () => { 
    this.setState({ quantity: this.state.quantity + 1 });
    this.updateInvoiceItemQuantity(this.props.invoiceId, this.state.quantity)
  }

  addFinalConsumer = () => {
    const { fcIdentification } = this.state;
    if (validateDni(fcIdentification)){
      const { fiscalIdentity, addFiscalIdentityToInvoice } = this.props;
      this.setLoading(true);
      addFiscalIdentityToInvoice('fc', fcIdentification, fiscalIdentity.id)
        .then(() => {
          this.setLoading(false);
          this.setShowCustomer(true)
        })
    } else {
      this.setState({validIdentity: false})
    }
  }

  renderViewItemsAdd = () => {
    return (
      <InvoiceItems
        items={this.props.items}
        total={this.props.invoiceTotal} 
        invoiceId ={this.props.invoiceId}
        incrementQuantity={this.incrementQuantity}
        quantity={this.state.quantity}
        deleteItem={this.props.deleteInvoiceItem}
      />
    ); 
  }

  renderCustomer = () => {
    return(
      <InvoiceCustomer 
        showCustomer={this.state.showCustomer}
        setShowCustomer={this.setShowCustomer}
        setFinalConsumer={this.setFcIdentification}
        identity={this.state.fcIdentification} 
        addFinalConsumer={this.addFinalConsumer}
        loading={this.state.loading}
        fiscalIdentity={this.props.fiscalIdentity}
      />
    );
  }

  render() {
    var date = presentInvoiceDate(this.state.invoiceDate);
    const { fiscalIdentity } = this.props;
    const { showCustomer, validIdentity } = this.state;
    const displayDni = validIdentity ? 'none' : 'flex';
    const typeCustomer = fiscalIdentity.name === 'fc' || !showCustomer ? 'Cosumidor Final' : 'Nombre Cliente';
    return(  
      <LinearGradient 
        colors={GRADIANTBLUE2} 
        style={style.container} 
        start={XY.startV} 
        end={XY.endV}>

        <View style={style.container}>
          <View style={style.containerBody}>
            
            <View style={style.containerReceptor}>
              <View style={style.inLineSpaceBetween}>
                <View style={style.boxVoucher}>
                  <Text style={[style.textRegular12White, {paddingBottom: 5}]}>
                    Tipo de Comprobante
                  </Text>
                  <Button
                    title={this.presentVoucherType()}
                    onPress={() => {this.setModalVisible(true)}}
                    buttonStyle={style.buttonVoucher}
                    titleStyle={style.textRegular16BlueCenter}
                  />
                </View>
                <View style={style.boxDate}>
                  <Text style={[style.textRegular12White, {paddingBottom: 5}]}>
                    Fecha de Emisión
                  </Text>
                  <Button
                    title={date}
                    onPress={this.showDateTimePicker}
                    buttonStyle={style.buttonDate}
                    titleStyle={style.textRegular16BlueCenter}
                  />
                  <DateTimePicker
                    isVisible={this.state.isDateTimePickerVisible}
                    onConfirm={this.handleDatePicked}
                    onCancel={this.hideDateTimePicker}
                    date={this.state.invoiceDate}
                  />
                </View>
              </View>
          
              <Text style={[style.textRegular12White, {paddingVertical: 5}]}>
                Datos del Receptor
              </Text>
              <View style={[style.containerCustomers,style.inColumnSpaceBetween]}>
                <View style={style.inLineSpaceBetween}>
                  <View style={style.textConsumerFinal}>
                    <Text style={style.textRegular16BlueMedium}>{typeCustomer}</Text>
                  </View>
                  <Button
                    title='Otro Cliente'
                    testID='addCustomer'
                    TouchableComponent={TouchableOpacity}
                    icon={IconAddCustomer}
                    onPress={ this.navigateClient }
                    buttonStyle={style.buttonAddCustomer}
                    titleStyle={style.textRegular12White}
                  />
                </View>
                
                <View style={{height: 4, left: 2, display: displayDni }}>
                  <Text style={style.textRegular12Red}>
                    Documento Inválido
                  </Text>
                </View>
                
                { this.renderCustomer() }
              
              </View>

              <Button
                title=' Agregar Producto/Servicio'
                TouchableComponent={TouchableOpacity}
                onPress={ this.navigateAddItems }
                icon={IconMore}
                buttonStyle={style.buttonAdd}  
                titleStyle={ style.textRegular16White }
                disabledStyle={style.buttonAddDisabled}
                disabledTitleStyle = { style.textRegular16GrayLight }
                ViewComponent={LinearGradient}
                linearGradientProps={COLORGBL}
              />

              <Text style={[style.textRegular12White, {paddingTop: 8}]}>
                Detalle Producto/Servicio
              </Text>

              { this.renderViewItemsAdd() }

            </View>
          </View>

          <View style={style.containerFooter}>
            <Button
              title='Continuar'
              TouchableComponent={TouchableOpacity}
              onPress={ this.navigateToSummaryInvoice }
              buttonStyle={style.buttonContinue}  
              titleStyle={ style.textRegular16White }
              ViewComponent={LinearGradient}
              linearGradientProps={COLORGBL}
            />
          </View>

          <Modal visible={this.state.modalVisible} animationType="fade" transparent={true}>
            <ModalVoucherTYpe selectionVoucher={this.selectionVoucher}/>
          </Modal>

        </View>
      </LinearGradient>
    )
  }
}

export default Invoice;