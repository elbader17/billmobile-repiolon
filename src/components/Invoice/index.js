import { connect } from 'react-redux';
import Invoice from './Component';
import { addFiscalIdentityToInvoice } from '../../app/fiscal_identity/actions';
import { updateInvoice, createInvoice } from '../../app/invoices/actions';
import { updateInvoiceItem } from '../../app/invoice_items/actions';

const mapStateToProps = state => ({
  fiscalIdentity: state.invoices.currentInvoice.fiscalIdentity,
  items: state.invoices.currentInvoice.invoiceItems,
  invoiceDate: state.invoices.currentInvoice.invoiceDate,
  voucherType: state.invoices.currentInvoice.voucherType,
  invoiceId: state.invoices.currentInvoice.id,
  invoiceTotal: state.invoices.currentInvoice.total,
});

const mapDispatchToProps = (dispatch) => {
  return {
    addFiscalIdentityToInvoice: (name, cuit, id) => (
      dispatch(addFiscalIdentityToInvoice(name, cuit, id))
    ),
    updateInvoiceItemQuantity: (id, quantity) => (
      dispatch(updateInvoiceItem(id, { quantity }))
    ),
    updateInvoice: values => (
      dispatch(updateInvoice(values))
    ),
    createInvoice: (invoiceDate, voucherType) => (
      dispatch(createInvoice(invoiceDate, voucherType))
    ),
  };
};


const component = connect(
  mapStateToProps,
  mapDispatchToProps,
)(Invoice);

export default component;
