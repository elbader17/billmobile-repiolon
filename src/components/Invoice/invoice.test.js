import React from 'react';
import { shallow, mount } from 'enzyme';
import Invoice from './Component';
import InvoiceCustomer from './InvoiceCustomer';

jest.mock('../../constants/metrics', () => ({ METRICS: { heightHeader: {} } }));
jest.mock('../../constants/invoice', () => ({ VOUCHER_TYPES: [{label:''}]  }));
jest.mock('../../utils/date', () => ({ presentInvoiceDate: () => {} }));
jest.mock('react-native-modal-datetime-picker')

const fnMock = jest.fn();

const props = {
  fiscalIdentity: {
    category: '',
    identification: '',
    name: ''
  },
  items: [],
  navigation:{navigate: fnMock},
}

test('render correctly component Invoice', () => {
  const component =  shallow(<Invoice {...props}/>);
  expect(component).toMatchSnapshot();
});

test('call the navigateToBeInvoice function', () => {
  const component = shallow(<Invoice {...props}/>);
  const button = component.findWhere(node => node.prop('testID') === 'continue');
  button.simulate('press');
  expect(fnMock).toBeCalled();
});

describe('test add final consumer', () => {
  test('check initial state cf is true', () => {
    const component = shallow(<Invoice {...props}/>);
    expect(component.instance().state.cf).toBe(false);
  });

  test('CONSUMER FINAL button set this state to true', () => {
    const component = shallow(<Invoice {...props}/>);
    const button = component.findWhere(node => node.prop('testID') === 'cf');
    button.simulate('press');
    expect(component.instance().state.cf).toBe(true);
  });

  test('CANCEL button set state cf to false', () => {
    const component = shallow(<Invoice {...props}/>);
    const button = component.findWhere(node => node.prop('testID') === 'cf');
    button.simulate('press');
    expect(component.instance().state.cf).toBe(true);
    const buttonCancel = component.findWhere(node => node.prop('testID') === 'cancel');
    buttonCancel.simulate('press');
    expect(component.instance().state.cf).toBe(false);
  });
});

test('call the navigateClient function', () => {
  const component = shallow(<Invoice {...props}/>);
  const button = component.findWhere(node => node.prop('testID') === 'addCustomer');
  button.simulate('press');
  expect(fnMock).toBeCalled();
});

test('call the navigateAddItems function', () => {
  const component = shallow(<Invoice {...props}/>);
  const button = component.findWhere(node => node.prop('testID') === 'addItems');
  button.simulate('press');
  expect(fnMock).toBeCalled();
});


