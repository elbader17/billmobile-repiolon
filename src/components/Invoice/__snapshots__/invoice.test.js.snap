// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly component Invoice 1`] = `
<ImageBackground
  source={
    Object {
      "testUri": "../../../src/images/gradiant.png",
    }
  }
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <ScrollViewMock>
    <View
      style={
        Object {
          "height": NaN,
          "paddingHorizontal": 18,
          "paddingVertical": 7,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "width": "60%",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "borderBottomColor": "#fff",
                "borderBottomWidth": 1,
                "borderRadius": 1,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "Lato-Regular",
                  "fontSize": 15,
                  "textAlign": "center",
                }
              }
            />
          </TouchableOpacity>
        </View>
        <View
          style={
            Object {
              "width": "35%",
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "borderBottomColor": "#fff",
                "borderBottomWidth": 1,
                "borderRadius": 1,
                "paddingVertical": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontFamily": "Lato-Regular",
                  "fontSize": 15,
                  "textAlign": "center",
                }
              }
            />
          </TouchableOpacity>
          <CustomDatePickerIOS
            cancelTextIOS="Cancel"
            confirmTextIOS="Confirm"
            hideTitleContainerIOS={false}
            isVisible={false}
            mode="date"
            neverDisableConfirmIOS={false}
            onCancel={[Function]}
            onConfirm={[Function]}
            onDateChange={[MockFunction]}
            onHideAfterConfirm={[MockFunction]}
            reactNativeModalPropsIOS={Object {}}
            titleIOS="Pick a date"
          />
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
              "borderRadius": 5,
              "elevation": 3,
              "marginVertical": 15,
              "maxHeight": 333.5,
            },
            Object {
              "flexDirection": "column",
              "justifyContent": "space-between",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
              Object {
                "margin": 7,
              },
            ]
          }
        >
          <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#C4C4C4",
                "borderWidth": 1,
                "height": 66.5,
                "width": 300,
              }
            }
            onPress={[Function]}
            testID="cf"
            title="CONSUMIDOR FINAL"
            titleStyle={
              Object {
                "color": "#858585",
                "fontFamily": "Lato-Regular",
                "fontSize": 11,
              }
            }
          />
          <ForwardRef(Themed.Button)
            TouchableComponent={[Function]}
            buttonStyle={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#C4C4C4",
                "borderWidth": 1,
                "height": 66.5,
                "width": 135,
              }
            }
            disabled={true}
            disabledStyle={
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              }
            }
            disabledTitleStyle={
              Object {
                "color": "transparent",
              }
            }
            onPress={[Function]}
            testID="cancel"
            title="Cancelar"
            titleStyle={
              Object {
                "color": "#C4C4C4",
                "fontFamily": "Lato-Regular",
                "fontSize": 12,
              }
            }
          />
          <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "#EE6123",
                "borderRadius": 3,
                "borderWidth": 1,
                "height": 66.5,
                "width": 105,
              }
            }
            icon={
              <Icon
                allowFontScaling={false}
                color="#EE6123"
                name="md-person-add"
                size={20}
              />
            }
            onPress={[Function]}
            testID="addCustomer"
          />
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#C4C4C4",
                "height": 1,
              },
              Object {
                "marginHorizontal": 5,
              },
            ]
          }
        />
        <InvoiceCustomer
          addFinalConsumer={[Function]}
          finalConsumer={false}
          fiscalIdentity={
            Object {
              "category": "",
              "identification": "",
              "name": "",
            }
          }
          loading={false}
          setFinalConsumer={[Function]}
        />
        <View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#C4C4C4",
                  "height": 1,
                },
                Object {
                  "marginHorizontal": 5,
                },
              ]
            }
          />
          <ForwardRef(Themed.Button)
            buttonStyle={
              Object {
                "backgroundColor": "transparent",
                "opacity": 0.8,
              }
            }
            title="VER TODOS"
            titleStyle={
              Object {
                "color": "#EE6123",
                "fontFamily": "Lato-Regular",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
      <ForwardRef(Themed.Button)
        buttonStyle={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 5,
            "elevation": 3,
            "height": 93.5,
          }
        }
        icon={
          <View
            style={
              Object {
                "position": "absolute",
                "right": 15,
              }
            }
          >
            <Icon
              allowFontScaling={false}
              color="#EE6123"
              name="md-add"
              size={30}
            />
          </View>
        }
        iconRight={true}
        onPress={[Function]}
        testID="addItems"
        title={
          <Text>
            <Text
              style={
                Object {
                  "color": "#858585",
                  "fontFamily": "Lato-Regular",
                  "fontSize": 14,
                }
              }
            >
              AGREGAR 
            </Text>
            <Text
              style={
                Object {
                  "color": "#858585",
                  "fontFamily": "Lato-Semibold",
                  "fontSize": 14,
                }
              }
            >
              ITEMS
            </Text>
          </Text>
        }
        titleStyle={
          Object {
            "color": "#858585",
            "fontFamily": "Lato-Regular",
            "fontSize": 14,
          }
        }
      />
      <View
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
          }
        }
      >
        <ForwardRef(Themed.Button)
          buttonStyle={
            Object {
              "backgroundColor": "#EE6123",
              "borderRadius": 0,
              "height": 106.5,
            }
          }
          disabled={true}
          disabledStyle={
            Object {
              "backgroundColor": "#C4C4C4",
              "borderRadius": 0,
              "height": 106.5,
            }
          }
          disabledTitleStyle={
            Object {
              "color": "#fff",
              "fontFamily": "Lato-Semibold",
              "fontSize": 14,
            }
          }
          onPress={[Function]}
          testID="continue"
          title="CONTINUAR"
          titleStyle={
            Object {
              "color": "#fff",
              "fontFamily": "Lato-Semibold",
              "fontSize": 14,
            }
          }
        />
      </View>
      <Component
        animationType="fade"
        hardwareAccelerated={false}
        transparent={true}
        visible={false}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgba(0,0,0,0.6)",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 3,
                "elevation": 20,
                "height": 533.5,
                "width": 525,
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#3687d1",
                  "borderTopLeftRadius": 3,
                  "borderTopRightRadius": 3,
                  "paddingVertical": 15,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontFamily": "Lato-Regular",
                    "fontSize": 15,
                    "textAlign": "center",
                  }
                }
              >
                Tipo de Comprobante
              </Text>
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                key="0"
              >
                <TouchableOpacity
                  activeOpacity={0.2}
                  onPress={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "borderColor": "#3687d1",
                        "borderRadius": 3,
                        "borderWidth": 1,
                        "paddingHorizontal": 10,
                        "paddingVertical": 3,
                      },
                      Object {
                        "marginVertical": 8,
                      },
                    ]
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#3687d1",
                        "fontFamily": "Lato-Regular",
                        "fontSize": 15,
                      }
                    }
                  />
                </TouchableOpacity>
              </View>
            </View>
          </View>
        </View>
      </Component>
    </View>
  </ScrollViewMock>
</ImageBackground>
`;
